refractor(location module) upgrade to 2x_style

jeremielodi opened the pull request:
fixing #3689

jeremielodi committed:
refractor(location module) upgrade to 2x_style

mbayopanda commented on client/src/modules/locations/country/country.html:
```diff
@@ -1,140 +1,44 @@
+  <div class="flex-content">
+    <div class="container-fluid">
+      <div
+        id="ountry-grid"
```
`country-grid`

mbayopanda commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
+  vm.createUpdateModal = (selectedCountry = {}) => {
+    return $uibModal.open({
+      templateUrl : 'modules/locations/country/modal/createUpdate.html',
+      controller : 'CountrysAddController as ModalCtrl',
```
The name of the controller is either : `CountryAddController` or `CountriesAddController`.
Since the controller is responsible of creation and update, you can choose a meaningful name in this way  : `CreateEditCountryController` or something else in this way

mbayopanda commented on client/src/modules/locations/country/modal/createUpdate.html:
```diff
@@ -0,0 +1,33 @@
+    </ol>
+  </div>
+  <div class="modal-body">
+    <div class="row">
```
Since you use the `.row` class, could you add the associated `.col-` class.

You can write something like : 
```html
<div class="row">
  <div class="col-xs-12">
    ...
  </div>
</div>
```

mbayopanda commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
+    cellTemplate : 'modules/locations/country/templates/action.cell.html',
+  }];
+
+  // ng-click="
```
You can remove this comment

jniles replied:
I agree with @mbayopanda on this one.  Go ahead and clean this up.

mbayopanda commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
+        Location.delete.country(uuid)
+          .then(() => {
+            Notify.success('FORM.INFO.DELETE_SUCCESS');
+            refreshCountrys();
```
Could you fix the function name `refreshCountries`

mbayopanda commented on client/src/modules/locations/locations.js:
```diff
@@ -23,12 +23,46 @@ function LocationController(locationService) {
+    headerCellFilter : 'translate',
+  }];
+
+  // ng-click="
```
Could you remove this comment

jniles replied:
Same comment.

mbayopanda commented on client/src/modules/locations/province/modal/createUpdate.js:
```diff
@@ -0,0 +1,46 @@
+angular.module('bhima.controllers')
+  .controller('ProvinceAddController', ProvinceAddController);
```
Is it possible to name the controller with something like `CreateEditProvinceController` or something else in this way.

mbayopanda commented on client/src/modules/locations/province/province.js:
```diff
@@ -3,93 +3,104 @@ angular.module('bhima.controllers')
+    cellTemplate : 'modules/locations/country/templates/action.cell.html',
+  }];
+
+  // ng-click="
```
Could you remove this comment

mbayopanda commented on client/src/modules/locations/sector/modal/createUpdate.js:
```diff
@@ -0,0 +1,57 @@
+angular.module('bhima.controllers')
+  .controller('SectorAddController', SectorAddController);
```
`CreateEditSectorController` or something else in this way

mbayopanda commented on client/src/modules/locations/sector/sector.html:
```diff
@@ -1,202 +1,46 @@
+  <div class="flex-content">
+    <div class="container-fluid">
+      <div
+        id="ountry-grid"
```
`sector-grid` i think, instead of `ountry-grid`

mbayopanda commented on server/config/routes.js:
```diff
@@ -179,6 +179,11 @@ exports.configure = function configure(app) {
+  app.delete('/locations/countries/:uuid', locations.delete.country);
+  app.delete('/locations/provinces/:uuid', locations.delete.province);
+  app.delete('/locations/sectors/:uuid', locations.delete.sector);
+  app.delete('/locations/villages/:uuid', locations.delete.village);
```
👍 excellent

Is it possible to have integration tests for these `delete` routes ?

mbayopanda reviewed (commented):
LGTM 👍 

I made just some comments that you can resolve before the merge

mbayopanda commented on client/src/modules/locations/village/modal/createUpdate.html:
```diff
@@ -0,0 +1,94 @@
+        <div class="col-xs-6">
+          <div class="form-group">
+            <label class="control-label" translate> FORM.LABELS.LONGITUDE </label>
+            <input type="number" step="1.0" placeholder="{{ 'FORM.PLACEHOLDERS.LONGITUDE' | translate }}"
```
It seems like it is not possible to save decimals values only integers

jniles reviewed (approved):
This looks fine to me.  There are a few improvements to make, but I'll save it for another issue.

jniles commented on test/end-to-end/locations/provinces.spec.js:
```diff
@@ -15,30 +16,58 @@ describe('Provinces Management', () => {
+  it('creates another province', async () => {
+
+    // switch to the create form
+    FU.buttons.create();
```
This needs an await in front of it.

mbayopanda commented on client/src/modules/locations/country/modal/createUpdate.html:
```diff
@@ -0,0 +1,33 @@
+          required="true">
+        </bh-input-text>
+      </div>
+    </div>
```
There is a `</div>` which is missing

mbayopanda commented on client/src/modules/locations/province/modal/createUpdate.html:
```diff
@@ -0,0 +1,46 @@
+          </div>
+        </div>
+      </div>
+    </div>
```
There is a `</div>` which is missing

mbayopanda commented on client/src/modules/locations/sector/modal/createUpdate.html:
```diff
@@ -0,0 +1,60 @@
+        </div>
+      </div>
+      </div>
+    </div>
```
There is a `</div>` which is missing here

mbayopanda commented on client/src/modules/locations/village/modal/createUpdate.html:
```diff
@@ -0,0 +1,94 @@
+    </ol>
+  </div>
+  <div class="modal-body">
+    <div class="row">
```
Since there is a `.row` here, you have to add a block with `.col-`. Something like : 
```html
<div class="row">
  <div class="col-xs-12">
  ...
  </div>
</div>
```

mbayopanda replied:
Also don't forget to correctly close `<div>` blocks with `</div>`.

jniles commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
 
 CountryController.$inject = [
-  'LocationService', 'util', 'NotifyService',
+  'LocationService', 'util', 'LocationService',
```
You can requested `LocationService` twice here.  Only once is needed.

mbayopanda reviewed (commented):
The feature works very well, i just add comments about some `<div>` blocks which are not correctly closed and another `.row` block without `.col-` child

jniles commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
+      resolve : { data : () => selectedCountry },
+    }).result.then(result => {
+      if (result) refreshCountries();
     });
```
If you click "cancel", a warning will probably appear in the console since you don't have  "catch" statement here.  You can fix this by adding `.catch(angular.noop)`.

jniles commented on client/src/modules/locations/locations.js:
```diff
@@ -23,12 +23,46 @@ function LocationController(locationService) {
+    enableSorting : true,
+    fastWatch : true,
+    flatEntityAccess : true,
+    onRegisterApi : (gridApi) => {
```
Is `gridApi` ever used?  You might want to use it to implement `toggleInlineFilter()` like in other modules.

jniles commented on client/src/modules/locations/locations.service.js:
```diff
@@ -59,6 +59,13 @@ function LocationService($http, util, Modal) {
   service.update.sector = updateSector;
   service.update.village = updateVillage;
 
+  service.delete = {
```
Nice!  👍

jniles commented on client/src/modules/locations/country/country.js:
```diff
@@ -2,91 +2,93 @@ angular.module('bhima.controllers')
+    enableSorting : true,
+    fastWatch : true,
+    flatEntityAccess : true,
+    onRegisterApi : (gridApi) => {
```
Do you ever need the `gridApi`?  I think you might have been wanting to use it for `toggleInlineFilter()` and just never implemented that function.  Go ahead and implement it.

jniles commented on client/src/modules/locations/province/province.js:
```diff
@@ -3,93 +3,104 @@ angular.module('bhima.controllers')
 
 ProvinceController.$inject = [
   'LocationService', 'util', 'NotifyService',
+  'ModalService', '$uibModal', 'uiGridConstants', 'LocationService',
```
LocationService is used twice here.

jniles commented on client/src/modules/locations/sector/sector.js:
```diff
@@ -3,118 +3,108 @@ angular.module('bhima.controllers')
 
 SectorController.$inject = [
   'LocationService', 'util', 'NotifyService',
+  'ModalService', '$uibModal', 'uiGridConstants', 'LocationService',
```
LocationService is used twice here.

jniles commented on server/controllers/admin/locations.js:
```diff
@@ -499,3 +515,37 @@ exports.update.village = function updateVillage(req, res, next) {
+  const sql = 'DELETE FROM country WHERE uuid=?';
+  const _uuid = db.bid(req.params.uuid);
+  db.exec(sql, _uuid).then(() => {
+    res.sendStatus(200);
```
According to our [API Design Guidelines](https://github.com/IMA-WorldHealth/bhima/wiki/Design:-Server-CRUD-APIs#api-design), the DELETE request success status code should be 204.

jniles commented on test/integration/locations.js:
```diff
@@ -261,4 +320,36 @@ describe('(/locations) Locations Interface', () => {
+  it('DELETE /locations/countries/:uuid should delete a Country', () => {
+    return agent.delete(`/locations/countries/${country2.uuid}`)
+      .then((res) => {
+        expect(res).to.have.status(200);
```
Go ahead and use `helpers.api.deleted(res)`.  It will make sure the API is conforming to the correct specification.

jniles reviewed (commented):
Just a few clean up things to improve the code legibility.  Nothing major here.

Well done!

jniles commented on client/src/modules/locations/province/province.js:
```diff
@@ -3,93 +3,104 @@ angular.module('bhima.controllers')
   const vm = this;
   vm.session = {};
   vm.view = 'default';
   vm.state = {};
```
One more thing - Please check that `vm.state` and `vm.default` are still used.  If they are not, please remove them from all the controllers.

mbayopanda replied:
@jeremielodi is it fixed ?

jniles commented on client/src/modules/locations/country/country.js:
```diff
@@ -3,90 +3,96 @@ angular.module('bhima.controllers')
-      ? locationService.create.country(country)
-      : locationService.update.country(country.uuid, country);
+  vm.remove = function remove(uuid) {
+    const message = 'FORM.DIALOGS.DELETE_ROLE';
```
These messages all say "DELETE_ROLE".  It doesn't make sense in this context.  Could you update the text?

![DeleteRoleOnProvincePage](https://user-images.githubusercontent.com/896472/63577368-20e9e600-c586-11e9-995b-e6f11740287e.PNG)

jniles replied:
@jeremielodi can you address this?  Then we can merge this code.

jniles commented on client/src/modules/locations/sector/modal/createUpdate.html:
```diff
@@ -0,0 +1,60 @@
+          ng-model="ModalCtrl.sector.country_uuid" id="country"
+          ng-change="ModalCtrl.loadProvinces()"
+          ng-options="c.uuid as c.name for c in ModalCtrl.countries | orderBy:'name'" required>
+          <option value="" disabled translate> {{ ModalCtrl.messages.country }} </option>
```
Does `{{ ModalCtrl.messages.country }}` exist?  I only get a blank text on most of these:

![Needs some text](https://user-images.githubusercontent.com/896472/63577432-48d94980-c586-11e9-8cc7-2ac32b7db2d3.PNG)

jniles reviewed (commented):
Just two more modifications and this looks good to me.

jniles committed:
fix(locations): "DELETE_ROLE" -> "DELETE_CONFIRM"

Cleans up the location deletion interface by using generic text instead
of "delete role" text.
